api/types/src/bytecode.rs

```diff
@@ -103,9 +103,17 @@ pub trait Bytecode {
                 mutable: true,
                 to: Box::new(self.new_move_type(t.borrow())),
             },
-            SignatureToken::Function(..) => {
-                // TODO
-                unimplemented!("signature token function to API MoveType")
+            SignatureToken::Function(args, result, abilities) => {
+                let new_vec = |toks: &[SignatureToken]| {
```

Comment:
nit: would this be more efficient as a separate function vs. building this lambda every time we are in this block?

Or alternatively, just use the iter().map().collect() on L113 and L114